web/src/routes/auth/login/+page.svelte

```diff
@@ -1,12 +1,21 @@
 <script lang="ts">
-  import { goto } from '$app/navigation';
+  import { afterNavigate, goto } from '$app/navigation';
   import LoginForm from '$lib/components/forms/login-form.svelte';
   import FullscreenContainer from '$lib/components/shared-components/fullscreen-container.svelte';
   import { AppRoute } from '$lib/constants';
   import { featureFlags, serverConfig } from '$lib/stores/server-config.store';
+  import { resetSavedUser } from '$lib/stores/user.store';
+  import { api } from '@api';
   import type { PageData } from './$types';
 
   export let data: PageData;
+
+  afterNavigate(async ({ from }) => {
+    if (from?.url.pathname === AppRoute.AUTH_CHANGE_PASSWORD) {
+      resetSavedUser();
+      await api.authenticationApi.logout();
+    }
```

Comment:
TBF this should probably be done during the change password process, no? It feels a bit weird to handle it here

Reply:
It is here to handle the case where the user navigate back, using the back button on the browser before changing the password

Reply:
Oh *before* changing the actual password... I see